### DISCUSSION TITLE
Berry script: controlling many rgb leds through 'tasmota.cmd' is slow and cant get integrated Leds library to work

### DISCUSSION THREAD

**Discussion by madmagic007** [arendst/Tasmota#21970] [Q&A]
To control rgb leds, i have to run `tasmota.cmd("LED" + str(pos) + " " + hex)`. For my TXUltimate, this is 28 leds, so 28 iterations, which take well over 300 ms to complete.
The [addressable LEDs](https://tasmota.github.io/docs/Berry_Addressable-LED/) doc page states a way where this could possibly be done without the use of running cmds for all of the leds, however i cant get this to work.
I believe the RGB digital signal comes from GPIO13, i only found that in this [esphome examples](url). Setting gpio13 or using the default, i cant figure out how to control the LEDs

exaple code used:
```berry
import animate
var strip = Leds(28, gpio.pin(gpio.WS2812, 0))
var anim = animate.core(strip)
```
or:
```berry
import animate
var strip = Leds(28, gpio.pin(13, 0))
var anim = animate.core(strip)
```

**Comment by sfromis** [comment]
The LED command supports setting multiple neopixel leds with a single command, just supply a list of color values, and a starting point. This is a **lot** faster than multiple commands, but if you do want multiple, you could also do it much faster with something like:
`tasmota.cmd(f'Backlog0 LED{pos1} {hex1};LED{pos2} {hex2}')`
(This also uses f-strings for easier string building than concatenations).

To use the animation framework, you need more than the initialization setup you posted, but also to "do something". Here's an example for a 5x5 matrix:
```
import animate
var strip = Leds(5 * 5, gpio.pin(gpio.WS2812, 0))
var anim = animate.core(strip)
anim.set_back_color(0x000000)
var pulse = animate.pulse(0xFF0000, 3, 2)
var osc1 = animate.oscillator(0, 23, 3000, animate.TRIANGLE)
osc1.set_cb(pulse, pulse.set_pos)
anim.start()
```

You can also set the colors on the `strip` object without needing an animator on top.
